frame/staking/src/pallet/mod.rs

```diff
@@ -782,6 +786,17 @@ pub mod pallet {
 			// and that MaxNominations is always greater than 1, since we count on this.
 			assert!(!T::MaxNominations::get().is_zero());
 
+			// ensure election results are always bounded with the same value
+			assert!(
+				<T::ElectionProvider as ElectionProviderBase>::MaxWinners::get() ==
+					<T::GenesisElectionProvider as ElectionProviderBase>::MaxWinners::get()
+			);
+			// ensure desired targets requested is always lower than max winners
+			// supported by the election provider.
+			// assert!(
+			// 	ValidatorCount::<T>::get() <=
+			// 		<T::ElectionProvider as ElectionProviderBase>::MaxWinners::get()
+			// );
```

Comment:
```suggestion
sp_io::TestExternalities::new_empty().execute_with(|| {
			assert!(
			 	ValidatorCount::<T>::get() <=
			 		<T::ElectionProvider as ElectionProviderBase>::MaxWinners::get()
			);
})
```

a better way to test this is in `fn build()`  of `GenesisBuilder`.

Reply:
and this should be added to the list of invariants of this pallet and should be in the try_state hook.

Reply:
Added this check to try state and genesis build.